src/test/javascript/spec/component/programming-exercise/programming-exercise-build-configuration.component.spec.ts

```
@@ -36,4 +43,54 @@
+        expect(comp.programmingExercise()?.buildConfig?.dockerFlags).toBe('[["network","none"],["env","key=value"]]');
+
+        comp.onDisableNetworkAccessChange({ target: { checked: false } });
+        expect(comp.programmingExercise()?.buildConfig?.dockerFlags).toBe('[["env","key=value"]]');
+
+        comp.onDisableNetworkAccessChange({ target: { checked: true } });
+        comp.onEnvVarsChange('');
+        expect(comp.programmingExercise()?.buildConfig?.dockerFlags).toBe('[["network","none"]]');
+    });
+
+    it('should parse existing docker flags', () => {
+        programmingExercise.buildConfig!.dockerFlags = '[["network","none"],["env","key=value"]]';
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)